src/passes/bytesConverter.ts

```diff
@@ -45,21 +45,22 @@ export class BytesConverter extends ASTMapper {
   }
 
   visitExpression(node: Expression, ast: AST): void {
-    const typeNode = safeGetNodeType(node, ast.compilerVersion);
-    if (!(typeNode instanceof IntLiteralType)) {
-      node.typeString = generateExpressionTypeString(replaceBytesType(typeNode));
+    const typeNode = safeGetNodeType(node, ast.inference);
+    if (typeNode instanceof IntLiteralType || typeNode instanceof StringLiteralType) {
+      return;
```

Comment:
Could you explain what is the purpose of `this.commonVisit(node, ast);` in methods like that?

Reply:
Honestly for me its the clearest solution. It explicitly states that literals are not affected by the method. It just filters them out immediately. Please also take a look at the [conversation](https://github.com/NethermindEth/warp/pull/813#discussion_r1064589050) where we've discussed other approaches.

Reply:
I'd even think of filtering them out by `node instanceof Literal` instead of using typenodes. WYT?

Reply:
Yes, I understand that, I'm just saying in terms of design I don't like stopping the execution at some node without visiting its children, when it should traverse the entire AST, unless of course, the logic of what is being done breaks if you do so, but that is not the case here. The other thing is what you mentioned in the last comment, if there are other types of literals, why are we only checking those?

Anyway, I checked `generateExpressionTypeString` and `replaceBytesType` methods and I think none of them modifies the typestring of these nodes. That said, my preferred solution would be to remove the condition completely and execute `node.typeString = generateExpressionTypeString(replaceBytesType(typeNode));` on all nodes. It should work the same way.

If you still think that the best solution is to use the `return`, then I do like what you said about filtering them out using `node instanceof Literal` but we need to check first that none of the typestrings a literal node might have are modified by `generateExpressionTypeString` and `replaceBytesType` methods.

Reply:
> The other thing is what you mentioned in the last comment, if there are other types of literals, why are we only checking those?

That is a good question. These nodes existed before and they weren't checked. It probably worked because - as you said - it doesn't matter if we filter them out or not. Anyway, since typestrings logic is going to be replaced soon anyway I don't think there is any sense in splitting hairs